grassroot-webapp/src/main/java/za/org/grassroot/webapp/controller/rest/group/GroupFetchController.java

```diff
@@ -433,4 +433,54 @@ public GroupFilterResponse filterGroupMembers(@RequestParam String groupUid,
         }
     }
 
+    @RequestMapping(value = "/province/municipalities",method = RequestMethod.GET)
+    @ApiOperation(value = "Loads the municipalities for the provided province")
+    public  ResponseEntity<List<Municipality>> getMunicipalities(@RequestParam String province){
+        logger.info("Province recieved = {}",province);
+        if(province.equals("undefined")){
+            return ResponseEntity.ok(Collections.emptyList());
+        }
+
+        locationInfoBroker.loadMunicipalityByCoordinates(29.151640,-25.260050);
+
+        Province province1 = Province.valueOf(province);
+        return ResponseEntity.ok(locationInfoBroker.getMunicipalitiesForProvince(province1));
+    }
+
+    @RequestMapping(value = "/members/location",method = RequestMethod.GET)
+    public ResponseEntity<Map<String,Municipality>> loadUsersWithLocation(@RequestParam String groupUid){
+        Group group = groupBroker.load(groupUid);
+        Set<String> memberUids = group.getMembers().stream().map(User::getUid).collect(Collectors.toSet());
+
+        List<UserLocationLog> userLocationLogs = locationInfoBroker.loadUsersWithLocationNotNUll(memberUids);
+
+        Map<String,Municipality> userMunicipalityMap = new HashMap<>();
+
+        for(UserLocationLog userLocationLog:userLocationLogs){
+            Municipality municipality =
+                    locationInfoBroker.
+                            loadMunicipalityByCoordinates(userLocationLog.getLocation().getLongitude(),userLocationLog.getLocation().getLatitude());
+
+            User user = getUser(group,userLocationLog.getUserUid());
+
+            userMunicipalityMap.put(user.getDisplayName(),municipality);
+        }
+
+        log.info("Municipalities for users with location = {}",userMunicipalityMap);
+
+        return ResponseEntity.ok(userMunicipalityMap);
+    }
+
+    private User getUser(Group group,String userUid){
```

Comment:
What is this method? This is going to be extremely inefficient. What is happening here? There is a method just on the group that will do this for you, and even that is a problem. Rethink this.

Reply:
Ooh, no this method i will delete it,i was testing something with it.